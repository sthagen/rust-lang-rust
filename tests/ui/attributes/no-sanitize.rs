#![feature(no_sanitize)]
#![feature(stmt_expr_attributes)]
#![deny(unused_attributes)]
#![allow(dead_code)]

fn invalid() {
    #[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
    {
        1
    };
}

#[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
type InvalidTy = ();

#[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
mod invalid_module {}

fn main() {
    let _ = #[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
    (|| 1);
}

#[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
struct F;

#[no_sanitize(memory)] //~ ERROR attribute should be applied to a function definition
impl F {
    #[no_sanitize(memory)]
    fn valid(&self) {}
}

#[no_sanitize(memory)]
fn valid() {}
