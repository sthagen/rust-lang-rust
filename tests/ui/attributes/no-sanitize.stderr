error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:7:5
   |
LL |       #[no_sanitize(memory)]
   |       ^^^^^^^^^^^^^^^^^^^^^^
LL | /     {
LL | |         1
LL | |     };
   | |_____- not a function definition

error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:13:1
   |
LL | #[no_sanitize(memory)]
   | ^^^^^^^^^^^^^^^^^^^^^^
LL | type InvalidTy = ();
   | -------------------- not a function definition

error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:16:1
   |
LL | #[no_sanitize(memory)]
   | ^^^^^^^^^^^^^^^^^^^^^^
LL | mod invalid_module {}
   | --------------------- not a function definition

error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:20:13
   |
LL |     let _ = #[no_sanitize(memory)]
   |             ^^^^^^^^^^^^^^^^^^^^^^
LL |     (|| 1);
   |     ------ not a function definition

error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:24:1
   |
LL | #[no_sanitize(memory)]
   | ^^^^^^^^^^^^^^^^^^^^^^
LL | struct F;
   | --------- not a function definition

error: attribute should be applied to a function definition
  --> $DIR/no-sanitize.rs:27:1
   |
LL |   #[no_sanitize(memory)]
   |   ^^^^^^^^^^^^^^^^^^^^^^
LL | / impl F {
LL | |     #[no_sanitize(memory)]
LL | |     fn valid(&self) {}
LL | | }
   | |_- not a function definition

error: aborting due to 6 previous errors

